homeassistant/components/smartthings/media_player.py

```diff
@@ -211,11 +230,35 @@ async def async_media_next_track(self) -> None:
 
     async def async_select_source(self, source: str) -> None:
         """Select source."""
-        await self.execute_device_command(
-            Capability.MEDIA_INPUT_SOURCE,
-            Command.SET_INPUT_SOURCE,
-            argument=source,
-        )
+        if self.supports_capability(Capability.SAMSUNG_VD_MEDIA_INPUT_SOURCE):
```

Comment:
Is it important to check `Capability.SAMSUNG_VD_MEDIA_INPUT_SOURCE` before `Capability.MEDIA_INPUT_SOURCE`?